SimMuon/GEMDigitizer/src/GEMPadDigiClusterProducer.cc

```diff
@@ -53,10 +53,15 @@ void GEMPadDigiClusterProducer::produce(edm::Event& e, const edm::EventSetup& ev
 }
 
 
-void GEMPadDigiClusterProducer::buildClusters(const GEMPadDigiCollection &det_pads, GEMPadDigiClusterCollection &out_clusters)
+void GEMPadDigiClusterProducer::buildClusters(const GEMPadDigiCollection &det_pads,
+                                              GEMPadDigiClusterCollection &out_clusters)
 {
+  // construct clusters
   for (const auto& ch: geometry_->chambers()) {
-    unsigned int nClusters = 0;
+
+    // proto collection
+    std::vector<std::pair<GEMDetId, GEMPadDigiCluster> > proto_clusters;
```

Comment:
@davidlange6 The selection I implemented (maximum first 8 in the list) will need to be optimized for sure. Hardware tests will have to show how to best select GEM/ME0 clusters in realistic conditions. The final selection will likely be very different. With this setup we can simulate the effects of various selection procedures. So in due time we can modify the algorithm to not produce an intermediate proto_collection